components/RatChecker/RatChecker.tsx

```diff
@@ -0,0 +1,118 @@
+    let ratFileInput = useRef(null)
+
+    function onFileUpload(e) {
+        setIsChecking(true)
+
+        var blobSlice = File.prototype.slice || (File.prototype as any).mozSlice || (File.prototype as any).webkitSlice,
+            file = e.target.files[0],
+            chunkSize = 2097152, // Read in chunks of 2MB
+            chunks = Math.ceil(file.size / chunkSize),
+            currentChunk = 0,
+            spark = new SparkMD5.ArrayBuffer(),
+            fileReader = new FileReader()
+
+        fileReader.onload = function (e) {
+            spark.append(e.target.result) // Append array buffer
+            currentChunk++
+
+            if (currentChunk < chunks) {
+                loadNext()
+            } else {
+                let hash = spark.end()
+                api.checkRat(hash).then(result => {
+                    setIsChecking(false)
+                    setCheckingResult(result)
+                    ratFileInput.current.value = ''
+                })
+            }
+        }
+
+        fileReader.onerror = function () {
+            console.warn('oops, something went wrong.')
+        }
+
+        function loadNext() {
+            var start = currentChunk * chunkSize,
+                end = start + chunkSize >= file.size ? file.size : start + chunkSize
+
+            fileReader.readAsArrayBuffer(blobSlice.call(file, start, end))
+        }
+
+        loadNext()
+    }
+
+    function getResultElement(): JSX.Element {
+        if (!checkingResult || isChecking) {
+            return null
+        }
+        if (checkingResult.rat.includes('No matching signature')) {
+            return (
+                <p style={{ color: 'white', fontSize: 'larger' }}>
+                    This mod file is not known. For further information check{' '}
+                    <a target="_blank" rel="noreferrer" href="https://isthisarat.com/">
+                        https://isthisarat.com/
+                    </a>
+                </p>
+            )
+        }
+        if (checkingResult.rat.includes('Yes')) {
+            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod file seems to be a rat. Be careful!</p>
+        }
+        if (checkingResult.rat.includes('No')) {
+            return <p style={{ color: 'lime', fontSize: 'larger' }}>This mod file doesn't seem to be a rat.</p>
+        }
+    }
+
+    return (
+        <>
+            <Card>
+                <Card.Header>
+                    <Card.Title>Rat Checker</Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    <p>
+                        Please only download the mod from our{' '}
+                        <a target="_blank" rel="noreferrer" href="https://discord.gg/wvKXfTgCfb">
+                            Discord
+                        </a>
+                        . The download link is in the #mod-releases channel.
+                    </p>
```

Comment:
```suggestion
```
This is for general rat checking, a link to our mod should only be a sideeffect not the first line.  
Also only download from the github https://github.com/coflnet/skyblockmod/releases